Server/mods/deathmatch/logic/luadefs/CLuaGenericDefs.cpp

```diff
@@ -26,3 +38,153 @@ bool CLuaGenericDefs::IsTransferBoxVisible()
+}
+
+bool CLuaGenericDefs::OutputServerLog(std::string_view message)
+{
+    CLogger::LogPrintf(LOGLEVEL_MEDIUM, "%.*s\n", message.length(), message.data());
+    return true;
+}
+
+bool CLuaGenericDefs::OutputScriptDebugLog(lua_State* luaVM, std::string message, std::optional<unsigned char> rawLevel, std::optional<unsigned char> rawColorR,
+                                           std::optional<unsigned char> rawColorG, std::optional<unsigned char> rawColorB)
+{
+    unsigned char level = rawLevel.value_or(3);
+
+    if (level > 4)
+        throw std::invalid_argument("level must be between 0 and 4");
+
+    unsigned char colorR = rawColorR.value_or(255);
+    unsigned char colorG = rawColorG.value_or(255);
+    unsigned char colorB = rawColorB.value_or(255);
+
+    switch (level)
+    {
+        case 0:
+            m_pScriptDebugging->LogDebug(luaVM, colorR, colorG, colorB, "%.*s", message.length(), message.c_str());
+            break;
+        case 1:
+            m_pScriptDebugging->LogError(luaVM, "%.*s", message.length(), message.c_str());
+            break;
+        case 2:
+            m_pScriptDebugging->LogWarning(luaVM, "%.*s", message.length(), message.c_str());
+            break;
+        case 3:
+            m_pScriptDebugging->LogInformation(luaVM, "%.*s", message.length(), message.c_str());
+            break;
+        case 4:
+            m_pScriptDebugging->LogCustom(luaVM, colorR, colorG, colorB, "%.*s", message.length(), message.c_str());
+            break;
+    }
+
+    return true;
+}
+
+bool CLuaGenericDefs::OutputConsole(std::string message, std::optional<CElement*> rawElement)
+{
+    CStaticFunctionDefinitions::OutputConsole(message.c_str(), rawElement.value_or(m_pRootElement));
+    return true;
+}
+
+bool CLuaGenericDefs::OutputChatBox(lua_State* luaVM, std::string message, std::optional<std::variant<CElement*, std::vector<CPlayer*>>> rawVisibleTo,
+                                    std::optional<unsigned char> rawColorR, std::optional<unsigned char> rawColorG, std::optional<unsigned char> rawColorB,
+                                    std::optional<bool> rawColorCoded)
+{
+    unsigned char colorR = 231, colorG = 217, colorB = 176;
+    bool          colorCoded = rawColorCoded.value_or(false);
+
+    if (rawColorR.has_value() && rawColorG.has_value() && rawColorB.has_value())
+    {
+        colorR = rawColorR.value();
+        colorG = rawColorG.value();
+        colorB = rawColorB.value();
+    }
+
+    std::vector<CPlayer*> sendList;
+
+    if (!rawVisibleTo.has_value() || std::holds_alternative<CElement*>(rawVisibleTo.value()))
+    {
+        CElement* element = (rawVisibleTo.has_value() ? std::get<CElement*>(rawVisibleTo.value()) : m_pRootElement);
+
+        if (IS_TEAM(element))
+        {
+            auto team = static_cast<CTeam*>(element);
+            sendList.resize(team->CountPlayers());
+            sendList.assign(team->PlayersBegin(), team->PlayersEnd());
```

Comment:
```cpp
sendList.reserve(team->CountPlayers());
sendList.insert(sendList.begin(), team->PlayersBegin(), team->PlayersEnd());
```

Reply:
```suggestion
            sendList.reserve(team->CountPlayers());
            sendList.insert(sendList.begin(), team->PlayersBegin(), team->PlayersEnd());
```
Converted to suggestion block :)